lib/constants/cycles.ts

```diff
@@ -1,58 +1,94 @@
 /**
- * Liturgical Day cycle that can be used as metadata for liturgical days.
+ * Represent the two main cycle, that compose the liturgical year in the Roman rite.
  */
 export const ProperCycles = {
+  /**
+   * The "Proper of Time" (or "Temporal") consists of the movable feasts,
+   * most of them keyed to Easter (which falls on a different Sunday every year),
+   * including Ascension, Pentecost, and so on.
+   */
   ProperOfTime: 'PROPER_OF_TIME',
+  /**
+   * The "Proper of Saints" (or "Sanctorale") consists of the fixed feasts,
+   * celebrated on the very same date each year (no matter what the day of the week),
+   * including Christmas and all the saints' days.
+   */
   ProperOfSaints: 'PROPER_OF_SAINTS',
 } as const;
 
 /**
- * Sundays cycle that can be used as metadata for liturgical days.
+ * A dynamically generated array from {@link ProperCycles} that contains all the possible proper cycles,
+ * that compose the liturgical year in the Roman rite.
  */
-export const SundaysCycles = {
+export const PROPER_CYCLES = Object.values(ProperCycles);
+
+/**
+ * Represent one of the two main cycles (available from {@link ProperCycles}),
+ * that compose the liturgical year in the Roman rite.
+ */
+export type ProperCycle = typeof PROPER_CYCLES[number];
+
+/**
+ * A three-year cycles for the Sunday mass readings (and some solemnities).
+ * The years are designated A, B, or C. Each yearly cycle begins on the first Sunday of Advent.
+ * Year B follows year A, year C follows year B, then back again to A.
```

Comment:
Well, the cycles are three, therefore neither `a three-year cycle` (one cycle lasting three consequetive years) nor `three-year cycles` (individual cycles each lasting three consecutive years) are true.

[_Yearly_](https://www.lexico.com/en/definition/yearly) mean _once a year_, _every year_, _annually_. We could use _year-long_ (adjective; some prefer writing it without hyphen), but I don’t it sounds good enough / naturally. Moreover, _Each cycle begins on the first Sunday of Advent._ tells us exactly how long each cycle takes.

[_[Holy] Mass_](https://www.lexico.com/en/definition/Mass) is always written with capital first letter in English.

> Year B follows year A, year C follows year B, then back again to A.

In my opinion, (1) this is too verbose; (2) you chose a complicated way to describe the order (at first sight, you have written `B` → `A`, `C` → `B`; one needs to read it more attentively to understand what you have writen).

```suggestion
 * There are three cycles for Sunday Mass readings (and some solemnities).
 * The cycles are designated A, B, or C. Each cycle begins on the First Sunday of Advent
 * of the previous civil year and ends on the Christ the King Solemnity.
 * The cycles follow each other in alphabetical order.
 * C year is always divisible by 3, A has remainder of 1, and B remainder of 2.
```

Reply:
Thank you for your suggestion (this sentence was a copy paste from Wikipedia...).

Reply:
A new suggestion (slightly modified from yours):

```suggestion
 * A three-year cycle for Sunday Mass readings (and some solemnities), designated by A, B, or C. 
 * Each cycle begins on the First Sunday of Advent of the previous civil year and ends on Saturday 
 * after the Christ the King Solemnity. The cycles follow each other in alphabetical order, 
 * starting from A in 1969 (the year when the new calendar and lectionary as been promulgated by Pope Paul VI).
```

Reply:
Okay, I have no problem with your latest suggestion, but do you really think that `starting from A in 1969 (the year when the new calendar and lectionary as been promulgated by Pope Paul VI)` is an important note in regards to `SundayCycles` constant? From my point of view, it is a nice historical fact, but knowing how can one get the Sunday cycle letter from a civil year is of much more use. :wink:

Reply:
Final suggestion:

```suggestion
 * A three-year cycle for Sunday Mass readings (and some solemnities), designated by A, B, or C. 
 * Each cycle begins on the First Sunday of Advent of the previous civil year and ends on Saturday 
 * after the Christ the King Solemnity. The cycles follow each other in alphabetical order.
 * C year is always divisible by 3, A has remainder of 1, and B remainder of 2.
```

Reply:
Thanks, LGTM.